hybrasyl/Objects/User.cs

```diff
@@ -2263,25 +2263,222 @@ public override void Attack(Castable castObject)
+                                        Map.EntityTree.GetObjects(new Rectangle(this.X - intent.Radius, this.Y, intent.Radius, 1))
+                                            .Where(
+                                                obj =>
+                                                    obj is Creature && obj != this && obj.GetType() != typeof(Merchant)));
+                                }
+                                    break;
+                                case Direction.East:
+                                {
+                                    //facing east, attack south
+                                    possibleTargets.AddRange(
+                                        Map.EntityTree.GetObjects(new Rectangle(this.X, this.Y - intent.Radius, 1, intent.Radius))
+                                            .Where(
+                                                obj =>
+                                                    obj is Creature && obj != this && obj.GetType() != typeof(Merchant)));
+                                }
+                                    break;
+                                case Direction.West:
+                                {
+                                    //facing west, attack north
+                                    possibleTargets.AddRange(
+                                        Map.EntityTree.GetObjects(new Rectangle(this.X, this.Y, 1, 1 + intent.Radius))
+                                            .Where(
+                                                obj =>
+                                                    obj is Creature && obj != this && obj.GetType() != typeof(Merchant)));
+                                }
+                                    break;
+                            }
+                        }
+                            break;
+                        case IntentDirection.Nearby:
+                        {
+                            //attack radius
+                            possibleTargets.AddRange(
+                                Map.EntityTree.GetObjects(new Rectangle(this.X - intent.Radius, this.Y,
+                                        (this.X + intent.Radius) - (this.X - intent.Radius),
+                                        (this.Y + intent.Radius) - (this.Y - intent.Radius)))
+                                    .Where(obj => obj is Creature && obj != this && obj.GetType() != typeof(Merchant)));
+                            possibleTargets.AddRange(
+                                Map.EntityTree.GetObjects(new Rectangle(this.X, this.Y - intent.Radius,
+                                        (this.X + intent.Radius) - (this.X - intent.Radius),
+                                        (this.Y + intent.Radius) - (this.Y - intent.Radius)))
+                                    .Where(obj => obj is Creature && obj != this && obj.GetType() != typeof(Merchant)));
+                        }
+                            break;

```

Comment:
I think my point is that all of these cases essentially define a rectangle to be used to query the quadtree - which is the only part that changes. So why not define only the rectangle in the `case` and have one call to `GetObjects/AddRange`? at the very end? It would reduce the length of this by a considerable amount.